src/spaces/homspace.jl

```diff
@@ -115,3 +115,16 @@ function dim(W::HomSpace)
     end
     return d
 end
+
+# Operations on HomSpaces
+# -----------------------
+function permute(W::HomSpace{S}, (p₁, p₂)::Index2Tuple{N₁,N₂}) where {S,N₁,N₂}
+    cod = ProductSpace{S,N₁}(map(n -> W[n], p₁))
+    dom = ProductSpace{S,N₂}(map(n -> dual(W[n]), p₂))
+    return cod ← dom
+end
+
+function Base.:*(W::HomSpace{S}, V::HomSpace{S}) where {S}
```

Comment:
I am not sure about this one. `HomSpace` is perhaps not a subtype of `VectorSpace`, but mathematically it kind of is, and on `VectorSpace` instances, `*` acts very differently.

Reply:
Ah, I forgot that we use `*` as a non-unicode otimes... What do you think about `\circ` instead? I don't really mind the name, it's just a simple convenience thing

Reply:
I changed it to just be `compose`. If we want, we can still add the unicode alternative, but I think this does not interfere with anything.

Reply:
I like compose; it might conflict with other packages but for now we don't export it I assume? I actually wouldn't mind also defining compose for `TensorMap`s. I think of `*` as a convenience symbol that does certain functionality depending on the type, i.e.
* `*` on vector spaces is `otimes`
* `*` on tensor maps is `compose`
* `*` on ... is ...